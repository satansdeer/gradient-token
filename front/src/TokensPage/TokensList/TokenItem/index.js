import React from "react";
import PropTypes from "prop-types";
import { PropTypes as MobxPropTypes } from "mobx-react";

import TokenImage from "components/TokenImage";
import "./TokenItem.css";

const TokenItem = ({ token, onClick }) => {
  return (
    <div className="TokenItem" onClick={onClick}>
      <div className="TokenItem-image_wrapper">
        <TokenImage outer={token[0]} inner={token[1]} />
      </div>
      <div className="TokenItem-label">{`${token[0]} â€“ ${token[1]}`}</div>
    </div>
  );
};

TokenItem.propTypes = {
  token: MobxPropTypes.arrayOrObservableArray,
  onClick: PropTypes.func
};

export default TokenItem;
