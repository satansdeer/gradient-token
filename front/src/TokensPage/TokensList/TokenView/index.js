import React from "react";
import { PropTypes as MobxPropTypes } from "mobx-react";
import TokenImage from "components/TokenImage";
import "./TokenView.css";

const TokenView = ({ token }) => (
  <div>
    <div className="TokenView-image_wrapper">
      <TokenImage size={200} outer={token[0]} inner={token[1]} />
    </div>
    <div className="TokenView-label">{`${token[0]} â€“ ${token[1]}`}</div>
  </div>
);

TokenView.propTypes = {
  token: MobxPropTypes.arrayOrObservableArray
};

export default TokenView;
