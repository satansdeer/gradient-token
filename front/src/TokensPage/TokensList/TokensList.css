.TokensList {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 30px 20px;
}

.TokensList-label_empty {
  font-size: 24px;
  color: #ccc;
}
