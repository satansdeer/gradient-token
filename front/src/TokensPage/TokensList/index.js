import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { PropTypes as MobxPropTypes } from "mobx-react";
import { inject, observer } from "mobx-react";

import TokenItem from "./TokenItem";
import TokenView from "./TokenView";
import "./TokensList.css";

const TokensList = ({
  gradientTokenStore: { tokens },
  modalStore: { showModal }
}) => {
  return (
    <Fragment>
      {tokens.length ? (
        <div className="TokensList">
          {tokens.map(token => (
            <TokenItem
              key={token.index}
              token={token.gradient}
              onClick={() => showModal(<TokenView token={token.gradient} />)}
            />
          ))}
        </div>
      ) : (
        <div className="TokensList-label_empty">You don't have tokens yet.</div>
      )}
    </Fragment>
  );
};

TokensList.propTypes = {
  gradientTokenStore: PropTypes.shape({
    tokens: MobxPropTypes.arrayOrObservableArrayOf(PropTypes.object)
  }),
  modalStore: PropTypes.shape({
    showModal: PropTypes.func
  })
};

export default inject("gradientTokenStore", "modalStore")(observer(TokensList));
