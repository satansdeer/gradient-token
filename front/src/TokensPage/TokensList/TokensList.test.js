import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import TokensList from "./index";

describe("<TokensList />", () => {
  const mockGradientTokenStore = {
    tokens: [{ index: 0, gradient: ["#fff", "#000"] }]
  };
  const mockModalStore = {
    modalStore: jest.fn()
  };

  it("renders and matches the snapshot", () => {
    const component = renderer.create(
      <TokensList.wrappedComponent
        gradientTokenStore={mockGradientTokenStore}
        modalStore={mockModalStore}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
