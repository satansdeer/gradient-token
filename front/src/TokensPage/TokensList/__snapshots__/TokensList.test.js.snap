// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokensList /> renders and matches the snapshot 1`] = `
<div
  className="TokensList"
>
  <div
    className="TokenItem"
    onClick={undefined}
  >
    <div
      className="TokenItem-image_wrapper"
    >
      <svg
        height={100}
        version="1.1"
        width={100}
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <radialGradient
            id="fff000"
          >
            <stop
              offset="0%"
              stopColor="#fff"
            />
            <stop
              offset="100%"
              stopColor="#000"
            />
          </radialGradient>
        </defs>
        <circle
          cx={50}
          cy={50}
          fill="url(#fff000)"
          r={46}
          stroke="#fff"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="TokenItem-label"
    >
      #fff â€“ #000
    </div>
  </div>
</div>
`;
